File: client/iih_stub/iih.ci_stub/iih.ci.diag/i/ICidiagQryService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using iih.bd.srv.diagdef.d;
using iih.bd.srv.ortpl.d;
using iih.ci.diag.cidiag.d;
using iih.ci.diag.dto.d;
using xap.sys.xbd.udi.d;

namespace iih.ci.diag_stub.i
{
    public interface ICidiagQryService
    {
        /// <summary>
        /// 
        /// </summary>
        /// <param name="id"></param>
        /// <param name="type"></param>
        /// <returns></returns>
        DIDTO[] QueryCiDiagDTO(string id, string type);
        DIDTO[] QuerRef(string id, string type);
        /// <summary>
        /// 基础诊断
        /// </summary>
        /// <param name="value"></param>
        /// <returns></returns>
        DiagDefDO[] getDiagDefDOS(String value);
        /// <summary>
        ///  诊断类型
        /// type  门诊  0 ，住院 1
        /// </summary>
        /// <param name="id_en"></param>
        /// <returns></returns>
        UdidocDO[] getDiType(string id_en, string type);
        /// <summary>string
        /// 参数  控制 科室是中医还是西医诊断
        /// </summary>
        /// <returns></returns>
        string getParamType();
        // 最新诊断
        CidiagAggDO[] getLastCiDiags(string id_en);
        //医嘱助手常用医嘱 分类
        /// <summary>
        /// 本次就诊的所有诊断
        /// </summary>
        /// <param name="id_en"></param>
        /// <returns></returns>
        CidiagAggDO[] getIdEntCiDiS(String id_en);

        /// <summary>
        /// 获取当前就诊对应的诊断明细
        /// </summary>
        /// <param name="id_en">就诊id</param>
        /// <param name="sd_ditp">诊断类型编码</param>
        /// <returns>本次就诊指定类型的诊断明细</returns>
        CiDiagItemDO[] getCiDiagItems(String id_en, String sd_ditp);
        /// <summary>
        /// 当前诊断
        /// </summary>
        /// <param name="id_en"></param>
        /// <param name="entp_code"></param>
        /// <returns></returns>
        CiDiagItemDO getCiDiagItemDO(String id_en, String entp_code);

        /// <summary>
        /// 获取本次就诊的保外诊断
        /// </summary>
        /// <param name="id_en"></param>
        /// <returns></returns>
        CiDiagItemDO[] getHpjudgetpCiDiagItems(String id_en);
    }
}

File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/diag/i/ICidiagQryService.java
package iih.ci.diag.i;

import iih.bd.srv.diagdef.d.DiagDefDO;
import iih.ci.diag.cidiag.d.CiDiagItemDO;
import iih.ci.diag.cidiag.d.CidiagAggDO;
import iih.ci.diag.dto.d.DIDTO;
import iih.ci.diag.dto.d.IpViewDiagDTO;
import xap.mw.core.data.BizException;
import xap.sys.xbd.udi.d.UdidocDO;

public interface ICidiagQryService {
	
	public abstract IpViewDiagDTO [] getCiDiagItemDOlist(String id_ent)throws BizException;
    
	/**
	 * getCiDiagDTO
	 * @param aggdos
	 * @throws BizException
	 */
	@Deprecated
    public abstract void getCiDiagDTO(DIDTO[] aggdos) throws BizException;
    
    /**
     * QueryCiDiagDTO
     * @param id
     * @return
     * @throws BizException
     */
	@Deprecated
    public abstract DIDTO[] QueryCiDiagDTO(String id,String type) throws BizException;
    
    /**
	  * 病案首页诊断显示
	  * @param id_ent
	  * @return
	  * @throws BizException
	  */
	 public abstract DIDTO[] getMedicalRecordDiag(String id_ent)throws BizException;
	 /**
	  * 取得疾病诊断 信息
	  * @param value 名称 拼音码 五笔码
	  * @return
	  * @throws BizException
	  */
	 public abstract DiagDefDO[] getDiagDefDOS(String value)throws BizException;
     /**
      * 诊断类型
      * param  id_en 就诊号
      * param  type  门诊 0    住院  1
      * @return UdidocDO【】
      * @throws BizException
      */
	 public abstract UdidocDO[] getDiType(String id_en,String type)throws BizException;
     /**
      * 诊断参数 中医 西医
      * @return
      * @throws BizException
      */
	 public abstract String getParamType()throws BizException;
	 
	 /**
	  * 根据 就诊查询所有诊断
	  * @return
	  * @throws BizException
	  */
	 @Deprecated
	 public abstract CidiagAggDO getCidiagAggDOIdEnAndDiTp(String id_en,String id_ditp) throws BizException;
     
	 /**
      * 根据就诊取得所有诊断信息
      * @param id_en
      * @return CidiagAggDO[]
      * @throws BizException
      */
      
	 public abstract CidiagAggDO[] getIdEntCiDiS(String id_en)throws BizException;
     /**
      *本次就诊的最新诊断
      * @param id_en
      * @return
      * @throws BizException
      */
     public abstract CidiagAggDO[] getLastCiDiags(String id_en) throws BizException;
     /**
      * 当前最新诊断的主诊断（id_en）
      * @param id_en
      * @param entp_code
      * @return
      * @throws BizException
      */
    public abstract CiDiagItemDO getCiDiagItemDO(String id_en,String entp_code)throws BizException;
    
    /**
     * 获取当前就诊对应的诊断
     * @param id_en 就诊id
     * @param sd_ditp 诊断类型编码
     * @return 本次就诊指定类型的诊断
     * @throws BizException
     */
    public CiDiagItemDO[] getCiDiagItems(String id_en, String sd_ditp) throws BizException ;
    
    /**
     * 获取当前就诊的保外诊断明细集合
     * @param id_en 就诊id
     * @return
     * @throws BizException
     */
    public CiDiagItemDO[] getHpjudgetpCiDiagItems(String id_en) throws BizException;
    
}
